sdcm/nemesis.py

```diff
@@ -5611,21 +5611,65 @@ def _finalizer(exc_type, *_):
+                    self.actions_log.info("Execute query on banned node")
+                    for key in random.sample(range(1, 100001), 1000):
                         stmt = SimpleStatement(f"INSERT INTO {keyspace_name}.{table_name} (key, name) VALUES ({key}, 'name{key}');",
                                                consistency_level=ConsistencyLevel.QUORUM)
-                        session.execute(stmt)
-                        self.log.error("Banned query passed to cluster from banned node")
-                        raise BannedQueryExecUnexpectedSuccess(
-                            "Query from banned node was executed succesful with Consistency.QUORUM")
-                    except (NoHostAvailable, OperationTimedOut, Unavailable) as exc:
-                        self.log.debug("Query failed with error: %s as expected", exc)
-                        self.actions_log.info("Query failed as expected")
+                        try:
+                            session.execute(stmt)
+                        except Exception as exc:  # noqa: BLE001
+                            self.log.info(
+                                "Query from banned node %s failed as expected: %r",
+                                self.target_node.name,
+                                exc,
+                            )
+                            break
+                        else:
+                            raise BannedQueryExecUnexpectedSuccess(
+                                "Query from banned node was executed successful with Consistency.QUORUM"
+                            )
+            except NoHostAvailable as exc:
+                self.log.info(
+                    "Could not establish CQL connection to banned node %s as expected: %s",
+                    self.target_node.name,
+                    exc,
+                )
+
+            self.actions_log.info(
+                "Wait for banned node %s to terminate after being removed from cluster", self.target_node.name
+            )
+
+            wait_for(
+                lambda: not self.target_node.db_up(),
+                step=5,
+                timeout=60,
+                throw_exc=True,
+                text=f"Wait banned node {self.target_node.name} to terminate after ban notification...",
+            )
+
+            patterns = [
+                "received notification of being banned from the cluster",
+                "terminating.",
+            ]
+
+            for pattern in patterns:
+                follower = self.target_node.follow_system_log(
+                    patterns=[pattern],
+                    start_from_beginning=True,
+                )
+                log_result = iter(follower)
+                if next(log_result, None) is None:
+                    raise AssertionError(
+                        f"Did not find expected log pattern {pattern!r} in system log of banned node {self.target_node.name}"
+                    )
+
+            self.log.info(
```

Comment:
Seems inconsistent, we are logging somethings into actions_logs and somethings into normal logs, should be unified